ui/app/templates/required_signoffs.html

```diff
@@ -0,0 +1,128 @@
+    </select>
+</form>
+</div>
+<h2>
+  <span>{{ selected_product }} Signoff Requirements</span>
+
+  <button class="btn btn-primary btn-xs" ng-click="addNewRequiredSignoff()">
+    Enable Signoffs for a new Product
+    <i class="glyphicon glyphicon-plus"></i>
+  </button>
+</h2>
+
+<div class="panel panel-default" ng-show="required_signoffs[selected_product]['permissions']">
+  <div class="panel-heading">
+    <h3 class="panel-title">
+      <span>
+        <span ng-show="state === 'current'">To change {{ selected_product}} permissions you must have signoff from...</span>
+          <button class="btn btn-xs btn-warning"
+                  ng-click="editRequiredSignoffs('permissions')">
+            Edit
+          </button>
+          <button class="btn btn-xs btn-danger"
+                  ng-click="deleteRequiredSignoffs(required_signoffs, 'permissions')">
+            Delete
+          </button>
+      </span>
+    </h3>
+  </div>
+  <div class="panel-body">
+    <!-- TODO: Would be great to put non-pending stuff at the top. -->
+    <div class="panel panel-default" ng-repeat="role in required_signoffs[selected_product]['permissions'] | toArray | orderBy: role.sc">
+      <div class="panel-heading">
+        <h4 class="panel-title">
+          <span ng-show="role['sc'] === null">{{ role["signoffs_required"] }} member(s) of {{ role.$key }}</span>
+          <span ng-show="role['sc'] !== null">{{ role["sc"]["signoffs_required"] }} member(s) of {{ role.$key }} (currently: {{ role["signoffs_required"] }})</span>
+          <span class="label label-info" ng-show="role['sc'] !== null && role['sc']['change_type'] !== 'delete'">Pending</span>
+          <span class="label label-danger" ng-show="role['sc'] !== null && role['sc']['change_type'] === 'delete'">Pending Delete</span>
+          <div style="float: right" ng-show="role['sc'] !== null">
+            <button class="btn btn-xs btn-primary" ng-click="signoff('permission', role['sc'], role.$key)"
+                    ng-show="! role['sc']['signoffs'].hasOwnProperty(current_user)">
+              Signoff as...
+            </button>
+            <button class="btn btn-xs btn-danger" ng-click="revokeSignoff('permission', role['sc'], role.$key)"
+                    ng-show="role['sc']['signoffs'].hasOwnProperty(current_user)">
+              Revoke your Signoff
+            </button>
+          </div>
+        </h4>
+      </div>
+      <div class="panel-body" ng-show="role['sc'] !== null">
+        <h4>Before it can be enacted, this change must be signed off on by:</h4>
+        <ul>
+          <li ng-repeat="(rs_role, signoffs_required) in role['sc']['required_signoffs']">
+            {{ signoffs_required }} members of {{ rs_role }}
+          </li>
+        </ul>
+        <h4 ng-show="role['sc'] !== null && ! isEmpty(role['sc']['signoffs'])">This change has already been signed off by:</h4>
```

Comment:
I agree that it's a bit weird that you can signoff on things that don't require you. I'll see if I can improve this. At the very least, maybe we can stop doing automatic signoffs if the users group is not required.